### PR TITLE
Doc updates for clustering support

### PR DESCRIPTION
**Pull request by jputrino**
@pjbreaux @swormke 

Fixes: #72, #73, #74 

#### What's this change do?
- Provides documentation for the new clustering templates
- Uses the new docs template for all f5_supported heat templates (which addresses the issue raised by @kmaloney )

#### Where should the reviewer start?
This is a large doc rebuild, so you'll need to build and view them locally (`sphinx-build docs docs/_build`). In particular, please look at the overview, prereqs, and config items for each template.

#### Any background context?
There is a supporting PR in the plugins repo that should be merged in first; that change needs to be built into the heat-plugins docs in order for multiple 'intersphinx' references to that doc set here to work.

Once this PR is merged into kilo, the changes need to be merged up to liberty and master. This should be done before the next release.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
     'sphinx.ext.intersphinx',
     'sphinx.ext.todo',
     'sphinx.ext.doctest',
+    'sphinx.ext.autosectionlabel'
```
Comment (pjbreaux) [inline docs/conf.py]: This appears to be causing the build failure.

File: docs/conf.py [hunk 2]
```diff
@@ -61,9 +62,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '7.0'
+version = "7.0.1"
 # The full version, including alpha/beta/rc tags.
-release = 'v7.0.1'
+release = "7.0.1"
```
Comment (pjbreaux) [inline docs/conf.py]: This will change for the release. v7.0.2

Comment (jputrino) [inline docs/conf.py]: I'd like to make this and the openstack version update automatically, like in the heat-plugins repo. Can we do that here?

**Comment by pjbreaux** [comment]
Looks good. Maybe minor changes in the future based on @szakeri feedback to templates.

**Pull request merged**